rumqttd/src/server/tls.rs

```diff
@@ -120,13 +120,13 @@ impl TLSAcceptor {
             #[cfg(feature = "use-rustls")]
             TLSAcceptor::Rustls { acceptor } => {
                 let stream = acceptor.accept(stream).await?;
-                let (_, session) = stream.get_ref();
-                let peer_certificates = session
-                    .peer_certificates()
-                    .ok_or(Error::NoPeerCertificate)?;
-                let tenant_id = extract_tenant_id(&peer_certificates[0].0)?;
+                // let (_, session) = stream.get_ref();
+                // let peer_certificates = session
+                //     .peer_certificates()
+                //     .ok_or(Error::NoPeerCertificate)?;
+                // let tenant_id = extract_tenant_id(&peer_certificates[0].0)?;
                 let network = Box::new(stream);
-                Ok((tenant_id, network))
+                Ok((None, network))
```

Comment:
We do want to have ability to get `tenant_id` , and getting it from certs is the best way ig. Is there any other way around it? 

If there or no certs, is it `None` or `Some([])` ? I think we can just check:
```rust
if !peer_certificates.is_empty() {
  // call extract here
}
// otherwise tenand_id will be None
```

cc: any comments @henil ?

Reply:
yeah `tenant_id` is only relevant when compiled with `validate-tenant-prefix` and we use this internally in our platform so i would like to keep it. We can also feature flag this logic so it doesn't run when broker is not compiled with `validate-tenant-prefix`.

> with native tls, it is commented out just like that as well
IIRC this was because at that time native-tls didn't support client side certificates

Reply:
Tenant id coming from cert is now behind `validate-tenant-prefix`.